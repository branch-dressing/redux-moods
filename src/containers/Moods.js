import React from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import { drinkCoffee, eatSnack, study, takeNap } from '../actions/actions';
import { useDispatch, useSelector } from 'react-redux';
import { getMood } from '../selectors/moodSelector';
import { startTimer } from '../actions/timerActions';

export const Moods = () => {
  const dispatch = useDispatch();
  const { reducer, timerReducer } = useSelector(getMood);
  const {coffees, snacks, naps, studies } = reducer;
  const { time } = timerReducer;
  
  return (
    <>
      <Controls>
        <button onClick={() => dispatch(startTimer())}>Time - {time}</button>
        <button onClick={() => dispatch(drinkCoffee())}>coffee - {coffees}</button>
        <button onClick={() => dispatch(eatSnack())}>snacks - {snacks}</button>
        <button onClick={() => dispatch(takeNap())}>naps - {naps}</button>
        <button onClick={() => dispatch(study())}>studies - {studies}</button>
      </Controls>
      <Face />
    </>
  );
};
