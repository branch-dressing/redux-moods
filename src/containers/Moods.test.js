import React from 'react';
import { shallow } from 'enzyme';
import Moods from './Moods';

describe('Moods', () => {
  describe.skip('container', () => {
    it('matches a snapshot', () => {
      const wrapper = shallow(<Moods />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
