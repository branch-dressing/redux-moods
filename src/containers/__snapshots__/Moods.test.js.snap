// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods container matches a snapshot 1`] = `<Moods />`;
